Automattic/wp-calypso
# Plans 2023: Woo $1 promotional pricing

Fixes Automattic/martech#2037

## Proposed Changes

Introduces the promotional/intro-offer pricing to the plans grid for the plans that have them.

- The current setup works only for site plans (derived from `/sites/[siteId]/plans`) with introductory offers added. When experiments are finalized, we'll determine how to put together the pricing, so some changes will be needed (discounts/pro-rated credits and add-ons will need to be taken into account).
- Billing descriptions added only for monthly & yearly plans to avoid more translations right now (added for both raw pricing and discounted pricing)

### Media

| monthly | yearly |
|---|---|
| <img width="950" alt="Screenshot 2023-09-01 at 1 11 58 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/1782d318-3622-42a0-8697-9f07f822edba"> | <img width="970" alt="Screenshot 2023-09-01 at 1 12 11 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/08cc97ec-6b5b-4c53-af62-fae00eb1b8db"> |

## Testing Instructions



* Sandbox public-api and disable store sandbox (more about this in phab diff D119416-code ) 
* Go to `/setup/wooexpress` and create a Woo Express site
* Go to `/plans/[woo-site]` and confirm pricing shows the intro offers
* Confirm `/start` renders as previously (no promo pricing)
* Confirm `/plans/[paid | free site]` renders as previously (no promo pricing)

## Pre-merge Checklist
